xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDecoder.h

```diff
@@ -69,15 +69,135 @@ namespace kodi
 namespace addon
 {
 
+//==============================================================================
+///
+/// \addtogroup cpp_kodi_addon_audiodecoder
+/// @brief \cpp_class{ kodi::addon::CInstanceAudioDecoder }
+/// **Audio decoder add-on instance**
+///
+/// Framework support for binary addons as audio decoder. This class implements a
+/// way to handle special types of audio files.
+///
+/// There is the complete way from loading the source, to outputting the usable
```

Comment:
The add-on handles loading of the source file and outputting the audio stream for consumption by the player.